sphinx/environment/__init__.py

```diff
@@ -358,17 +357,16 @@ def merge_info_from(self, docnames, other, app):
         app.emit('env-merge-info', self, docnames, other)
 
     def path2doc(self, filename):
-        # type: (unicode) -> unicode
+        # type: (unicode) -> Optional[unicode]
```

Comment:
why is `filename` optional?

Reply:
That's the return value, not the `filename` argument - and `Optional[unicode]` means that it may return a unicode string *or* `None` (if no suffix matches).

Reply:
oh, sorry. Please forget my comment...